src/odemis/acq/test/stream_test.py

```diff
@@ -3565,14 +3594,57 @@ def on_im(im):
             if im is not None:
                 e.set()
 
-        ars.image.subscribe(on_im)
+        ars.image.subscribe(on_im)  # when .image VA changes, call on_im(.image.value)
         e.wait()
+
+        im2d3 = ars.image.value
+        time.sleep(0.5)  # wait shortly as .image is updated multiple times
+
         # corresponding bg image
         bg_data = [model.DataArray(numpy.ones((1, 1, 1, 512, 1024), dtype=numpy.uint16), metadata[0])]
         e.clear()
         ars.background.value = bg_data
         # test if bg VA shows same values as stored in bg_data
-        numpy.testing.assert_equal(ars.background.value[0], bg_data[0][0, 0, 0])
+        numpy.testing.assert_array_equal(ars.background.value[0], bg_data[0][0, 0, 0])
+        e.wait()
+
+        im2d4 = ars.image.value
+        time.sleep(0.5)  # wait shortly as .image is updated multiple times
+        # check if the bg image has been applied and the .image VA has been updated
+        assert not numpy.array_equal(im2d3, im2d4)
+
+        ###################################################################
+        # test 1 large image + corresponding bg image to test resizing
+        data = [model.DataArray(1500 + numpy.zeros((2000, 2200), dtype=numpy.uint16), metadata[0])]
+        data[0][200:250, 50:70] = 1000  # modify a few px close to AR_POLE
+        logging.info("setting up stream")
+        ars = stream.StaticARStream("test arpol static stream with large image", data)
+
+        # wait a bit for the image to update
+        e = threading.Event()
```

Comment:
You don't need to redefine `threading.Event` and `om_im()` in every part of the test case. That said, it seems like this test case could be broken up into three test cases, as each part is independent of the other ones. Could you do of these 2 alternatives?